### DISCUSSION TITLE
which way better to use object value as hook's deps?

### DISCUSSION THREAD

**Discussion by lamualfa** [pmndrs/valtio#433] [Q&A] [answered]
## declaring the `state` & `snap`

```js
const state = useConstant(() => proxy({
  user: {
    name: 'lorem',
    email: 'ipsum'
  }
}))
const snap = useSnapshot(state)
```

## 1. passing the entire `snap`

```js
useEffect(() => {
  console.log(state.user)
}, [snap])
```

## 2. passing the specific field of the `snap`

```js
useEffect(() => {
  console.log(state.user)
}, [snap.user])
```

## 3. passing the specific value of the `snap.user`

```js
useEffect(() => {
  console.log(state.user)
}, [snap.user.name, snap.user.email])
```

**Comment by dai-shi** [comment, marked as answer]
The third one is most preferred.
deps (and memo) is very tricky for valtio. The safest way is to use only non-object values such as strings and numbers.
